components/concept/fetch/src/main/java/mozilla/components/concept/fetch/Request.kt

```diff
@@ -61,6 +65,23 @@ data class Request(
              * Create a [Body] from the provided [File].
              */
             fun fromFile(file: File): Body = Body(file.inputStream())
+
+            /**
+             * Create a [Body] from the provided [unencodedParams] in the format of Content-Type
+             * "application/x-www-form-urlencoded". Parameters are formatted as "key1=value1&key2=value2..."
+             * and values are percent-encoded. If the given map is empty, the response body will contain the
+             * empty string.
+             *
+             * @see [Headers.Values.CONTENT_TYPE_FORM_URLENCODED]
+             */
+            fun fromParamsForFormUrlEncoded(unencodedParams: Map<String, String>): Body {
```

Comment:
nit: Another option instead of a `Map` parameter would be: `vararg params: Pair<String, String>` (and internally do `params.toMap()`.

This allows you to not have to create a Map when calling the method:

```kotlin
Request.Body.fromParamsForFormUrlEncoded(
    "v" to "2",
    "url" to inputUrl
)
```

Reply:
Not strictly needed and could be a follow-up. Approving and merging this as-is. :)

Reply:
Filed https://github.com/mozilla-mobile/android-components/issues/2479